syntax="proto2";
package hit_consistency;
option cc_generic_services = true;

message StateMachineRequest {
    required int64 offset = 1;
    optional int32 size = 2;
}

message StateMachineResponse {
    required bool success = 1;
    optional string redirect = 2;
}

service StateMachineService {
    rpc write(StateMachineRequest) returns(StateMachineResponse);
    rpc read(StateMachineRequest) returns(StateMachineResponse);
};
